exo-run/src/service-restarter.ls

```diff
@@ -0,0 +1,52 @@
+require! {
+  'chokidar' : {watch}
+  'events' : {EventEmitter}
+  '../../exosphere-shared' : {DockerHelper}
+  'path'
+}
+
+
+# Watches local services for changes and restarts them
+class ServiceRestarter extends EventEmitter
+
+  ({@role, @service-location, @env, @logger}) ->
+    @docker-config-location = path.join process.cwd!, 'tmp'
+
+
+  watch: ~>
+    /* Ignores any sub-path including dotfiles.
+    '[\/\\]' accounts for both windows and unix systems, the '\.' matches a single '.', and the final '.' matches any character. */
+    @watcher = watch @service-location, ignore-initial: yes, ignored: [/.*\/node_modules\/.*/, /(^|[\/\\])\../]
+      ..on 'add', (added-path) ~>
+        @logger.log {role: 'exo-run', text: "Restarting service '#{@role}' because #{added-path} was created"}
+        @_restart!
+      ..on 'change', (changed-path) ~>
+        @logger.log {role: 'exo-run', text: "Restarting service '#{@role}' because #{changed-path} was changed"}
+        @_restart!
+      ..on 'unlink', (removed-path) ~>
+        @logger.log {role: 'exo-run', text: "Restarting service '#{@role}' because #{removed-path} was deleted"}
+        @_restart!
+
+
+  _restart: ->
+    @watcher.close!
+    cwd = @docker-config-location 
+    DockerHelper.kill-container {service-name: @role, cwd, @write}, (exit-code) ~>
+      | exit-code => @emit 'error', "Docker failed to kill container #{@role}"
+      @write "Docker container stopped"
```

Comment:
I don't know if we have style preferences for this, but don't we usually prefer single quotes to double, when possible? If so, same would go for line 40.